global/scripts/MyelinMap_BC.sh

```diff
@@ -0,0 +1,188 @@
+		fi
+        ;;
+esac
+
+case "$RegName" in
+	("")
+		RegNameInOutputName=""
+		RegNameInT1wName=""
+		RegNameStructString=""
+		;;
+	(MSMSulc)
+		RegNameInOutputName=_${RegName}
+		RegNameInT1wName=""
+		RegNameStructString=.${RegName}
+		;;
+	(*)
+		RegNameInOutputName=_${RegName}
+		RegNameInT1wName=_${RegName}
+		RegNameStructString=.${RegName}
+        ;;
+esac
+log_Msg "RegNameInOutputName: $RegNameInOutputName"
+log_Msg "RegNameInT1wName: $RegNameInT1wName"
+log_Msg "RegNameStructString: $RegNameStructString"
+
+NativeMyelinMap=${NativeFolder}/${Subject}.MyelinMap.native.dscalar.nii
+log_File_Must_Exist "${NativeMyelinMap}"
+
+IndividualLowResMap=${LowResFolder}/${Subject}.MyelinMap${RegNameInOutputName}.${LowResMeshString}.dscalar.nii
+${Caret7_Command} -cifti-resample ${NativeMyelinMap} \
+	COLUMN ${LowResFolder}/${Subject}.MyelinMap.${LowResMeshString}.dscalar.nii \
+	COLUMN ADAP_BARY_AREA ENCLOSING_VOXEL \
+	${IndividualLowResMap} \
+	-surface-postdilate 40 \
+	-left-spheres ${NativeFolder}/${Subject}.L.sphere${RegNameStructString}.native.surf.gii ${LowResFolder}/${Subject}.L.sphere.${LowResMeshString}.surf.gii \
+	-left-area-surfs ${NativeT1wFolder}/${Subject}.L.midthickness.native.surf.gii ${LowResFolder}/${Subject}.L.midthickness.${LowResMeshString}.surf.gii \
+	-right-spheres ${NativeFolder}/${Subject}.R.sphere${RegNameStructString}.native.surf.gii ${LowResFolder}/${Subject}.R.sphere.${LowResMeshString}.surf.gii \
+	-right-area-surfs ${NativeT1wFolder}/${Subject}.R.midthickness.native.surf.gii ${LowResFolder}/${Subject}.R.midthickness.${LowResMeshString}.surf.gii
+
+log_Msg "Resampled MyelinMap in the low res mesh space using registration: ${IndividualLowResMap}"
+
+ReferenceMap=${LowResFolder}/${Subject}.${LowResBCMapName}_BC.${LowResMeshString}.dscalar.nii
+log_File_Must_Exist "${ReferenceMap}"
+ReferenceMapOld=${LowResFolder}/${Subject}.${LowResBCMapName}_oldBC.${LowResMeshString}.dscalar.nii
+ReferenceMapToUse=${ReferenceMap}
+
+# TODO: if _oldBC exists, then _BC is from the new method; if _oldBC doesn't exists, then move forward to the if block.
+# match the reference map's mean with the individual map's mean
+if ((UseIndMeanBool)); then
+	log_Msg "match the reference map's mean with the individual map's mean"
+	log_Msg "UseIndMeanBool: $UseIndMeanBool"
+	# the new reference map
+	# TODO: Do we want to distinguish the mean-matched reference map by inserting a string in the file name?
+	ReferenceMapMeanMatch=${LowResFolder}/${Subject}.${LowResBCMapName}_BC.${LowResMeshString}.dscalar.nii
+	# calcualte means of reference and individual maps
+	# not average in vertex areas, but it is fine
+	MeanRef=$(${Caret7_Command} -cifti-stats -reduce MEAN ${ReferenceMap})
+	MeanInd=$(${Caret7_Command} -cifti-stats -reduce MEAN ${IndividualLowResMap})
+	log_Msg "MeanRef: $MeanRef"
+	log_Msg "MeanInd: $MeanInd"
+	# to save the old maps as _oldBC
+	if [ -f "$ReferenceMap" ]; then
+		cp ${ReferenceMap} ${ReferenceMapOld}
```

Comment:
If we are going to overwrite this anyway, mv is much faster than cp.  I thought we were wanting to preserve the old BC myelin map, not the old resampled group reference.

I'm also not sure the script itself should have anything to do with saving the old version, as running it twice will blow it away.  That should probably be a separate job, and this script shouldn't mess with the oldBC filenames at all.